specs/flyout/flyout.md

```diff
@@ -0,0 +1,168 @@
+# Flyout
+
+## Overview
+Flyout is a focusable floating container that shows over a pages content and displays UI related to what the user is doing. Similar to a Dialog, but with positioning controls and no specific action is required, a Flyout can be used to reveal a secondary control or show more detail about an item.
+
+### Background
+A Flyout can be less intrusive to the user experience than a Dialog and should be used instead of a Dialog when information or input is needed to be displayed, but not in the center of the screen, such as in the case of collection of additional info before an action, displaying information that is only relevant some of the time, or displaying more information.
+
+### Use Cases
+- As a Dropdown for a menu, navigation bar, etc.
+- In a context menu as the secondary menu/action.
+- As a popup showing additional info for an item on the page.
+- To display a larger image of a thumbnail image in a photo gallery.
+- To display an action related to a page item
+
+### Features
+**Placement Logic:** - This component will use *Anchored Region* to be positioned relative to the Flyouts anchor element. This way it can be dynamic and adapt to the available space if `responsive` is true for the Flyout.
+
+**Light Dismiss (aka Soft Dismiss):** - A way to quickly close the Flyout by clicking outside it, or pressing the `esc` key.
+
+**Overlay:** - A control of whether or not the Flyout has an overlay effect that dims the surrounding UI when the Flyout is active.
+
+### Prior Art/Examples
+##### Context Menu
+*Note:* the "Manage, Apply, Show more options" is the Flyout for this example.
+![Context Menu](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/flyout-nested.png)
+
+##### Text Only
+*Note:* the button is not part of the Flyout, but shown for context as an anchor element.
+![Text Only](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/flyout-wrapping-text.png)
+
+##### Actionable
+*Note:* the button is not part of the Flyout, but shown for context as an anchor element.
+![Actionable More Info](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/flyout-example2.png)
+
+---
+
+### API
+*Component Name*
+- `fast-flyout`
+
+*Attributes*
+- `anchor` - string, the html id of the HTMLElement that the Flyout is attached to, (viewport for anchored region defaults to the parent of the anchor)
+- `overlay` - boolean, specifies whether an overlay is drawn to dim the out of scope UI until the Flyout is dismissed
+- `light-dismiss` - boolean, specifies whether or not the flyout can be dismissed by clicking elsewhere or pressing `esc` key, defaults to true
```

Comment:
I sometimes see this called "soft dismiss" - did you happen to look at other libraries to see what is more common?

Reply:
I think there's been some discussion or usage of terminology over in Open UI. Might be good to dig into issues and PR's to see how they've referred to it. In Dialog, we emit an event regardless in the case of light dismiss rather than having at attribute here. Would that be preferred here as well? I think the benefit is that if you want to do something on dismiss you listen for the event; if you don't want to - you don't. Thoughts on creating alignment between these two?